### ISSUE TITLE
pendant cam

### ISSUE THREAD

**Issue by Mgilbride** [winder/Universal-G-Code-Sender#191]
Not to be completely redonkulous but, a webcam feed would rock. Would allow you to walk away from the machine anb still monitor and stop / pause a job. Looking forward to playing with this.

**Comment by pelrun** [comment]
I wrote a simple webcam plugin over here: https://github.com/pelrun/ugs-platform-plugin-camera

It includes crosshairs for doing alignment/registration, an appropriately configured macro can then set the appropriate reference point (e.g. I'm using G10 L20 with the offset of the camera from the tool head)

**Comment by winder** [comment]
@pelrun there aren't many people making UGS plugins, this made my day.

**Comment by pelrun** [comment]
UGS does nearly everything already...

**Comment by vMeph** [comment]
@pelrun could you explain how to install plug in and how it works?

**Comment by pelrun** [comment]
I added a README, but just download the nbm from the release tab and install it from Tools->Plugins. It'll appear in Window->Plugins->Camera.

**Comment by vMeph** [comment]
@pelrun whats the name of file?
 i go to tools- plugins-download-add plugin and when i open to add a .nbm file i see no file to add in, i have search in all folders and nothing

**Comment by vMeph** [comment]
ok i found my bad, its in releases the file i was downloading the source
now on the instalation im getting this
![1](https://user-images.githubusercontent.com/12812111/29835780-e8adc41c-8cea-11e7-9a63-7724997ad7b4.png)

**Comment by winder** [comment]
@vMeph The auto-update stuff I added recently is messing this up a bit. Basically since @pelrun just rebuilt everything the version is newer than last nights nightly build.

I just reran the build, so if you auto-update (or download the nightly) it should work now.

**Comment by pelrun** [comment]
To be honest I think my plugin is actually a completely neutral Netbeans Platform plugin (I'm not actually *using* anything in ugslib) so I might be able to remove the dependency entirely... (!) A job for tomorrow though.

@vMeph: since my plugin is separate to UGS, it's probably better if you raise any issues you have with it on my repository directly rather than here.

**Comment by winder** [comment]
@pelrun yes that's correct. Depending on `UGS Lib` allowed you to import the `BackendAPI` and `Settings` objects initialized in your constructor.

If you wanted to you could use that object to call `G10 L20 P0 X<offset_x> Y<offset_y> Z<offset_Z>`. Or you could remove it and install the plugin in the Netbeans IDE.

**Comment by vMeph** [comment]
thanks i have tested i must say looking very good works perfect, it was out of date @winder 

@pelrun i was wondering if you ever tryed this icon on UGS
![2](https://user-images.githubusercontent.com/12812111/29837396-ee59ead0-8cef-11e7-8072-aa5b070da140.png)
if something like this could be implemented on a app like that, that you can be on a smartphone and control what machine is doing, that could be a amazing app
unfortunatly that app have some problems that needs to be solved and dont have a webcam option

great job you did it here!!

**Comment by pelrun** [comment]
@winder: that was my initial plan, but I wasn't particularly happy with the resulting UI. Especially once I realised it would have just been a less-capable copy of the existing macro functionality, and I ripped it all out.

**Comment by Mgilbride** [comment]
Awesome! Looking forward to trying it.
